globus_action_provider_tools/flask/apt_blueprint.py

```diff
@@ -153,6 +179,47 @@ def action_introspect(self):
 
         return jsonify(self.provider_description), 200
 
+    def action_enumerate(self, func: ActionEnumerationType):
+        """
+        Decorates a function to be run as an Action Provider's enumeration
+        endpoint.
+        """
+
+        @functools.wraps(func)
+        def wrapper():
+            if not g.auth_state.check_authorization(
+                self.provider_description.runnable_by,
+                allow_public=True,
+                allow_all_authenticated_users=True,
+            ):
+                current_app.logger.info(
+                    f'User "{g.auth_state.effective_identity}" is unauthorized to enumerate Actions'
+                )
+                raise NotFound
+
+            valid_statuses = set(e.name.lower() for e in ActionStatusValue)
```

Comment:
I switched over to using casefold() just because it's name sorta implies that it's trying to do some case-insenstive comparisons.  The case insenstive checks aren't perfect but Enums names are kinda annoying to work with.